### PR TITLE
Updated anno_img

### PR DESCRIPTION
**Pull request by dakomura**
Following issue #64, I have updated `anno_img`.

> (1). add some space in the boundaries of two columns/rows?

Boders can now be inserted between images. The thickness of these border lines is controlled by `border_width`, and their color is defined by `border_color` (where 0 is black and 255 is white).

> (2). Test whether the row_split_gap or col_split_gap shows as expected when row_split or col_split are specified.

I verified to work as expected.

> (3) Add plot_kws, such as camp, norm, vmax, vmin that will be passed to ax.imshow

I have added `plot_kws`. But most of the parameters such as cmap, norm, vmax, and vmin are not applicable when the input image is in RGB color.

> (4). Other useful parameters or functions.

As mentioned in (1), parameters for borderlines have been added to enhance functionality.

The sample code and result are as follows.

```
df = pd.DataFrame(['AAAA1'] * 5 + ['BBBBB2'] * 5, columns=['AB'])
df['CD'] = ['C'] * 3 + ['D'] * 3 + ['G'] * 4
df['F'] = np.random.normal(0, 1, 10)
df.index = ['sample' + str(i) for i in range(1, df.shape[0] + 1)]
df_box = pd.DataFrame(np.random.randn(10, 4), columns=['Gene' + str(i) for i in range(1, 5)])
df_box.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar = pd.DataFrame(np.random.uniform(0, 10, (10, 2)), columns=['TMB1', 'TMB2'])
df_bar.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_scatter = pd.DataFrame(np.random.uniform(0, 10, 10), columns=['Scatter'])
df_scatter.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar1 = pd.DataFrame(np.random.uniform(0, 10, (10, 2)), columns=['T1-A', 'T1-B'])
df_bar1.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar2 = pd.DataFrame(np.random.uniform(0, 10, (10, 2)), columns=['T2-A', 'T2-B'])
df_bar2.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar3 = pd.DataFrame(np.random.uniform(0, 10, (10, 2)), columns=['T3-A', 'T3-B'])
df_bar3.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar3.iloc[5,0]=np.nan
df_bar4 = pd.DataFrame(np.random.uniform(0, 10, (10, 1)), columns=['T4'])
df_bar4.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_bar4.iloc[7,0]=np.nan
df_img = pd.DataFrame([f"{i:02}.jpg" for i in range(1,11)], columns=['path'])
df_img.index = ['sample' + str(i) for i in range(1, df_box.shape[0] + 1)]
df_heatmap = pd.DataFrame(np.random.randn(30, 10), columns=['sample' + str(i) for i in range(1, 11)])
df_heatmap.index = ["Fea" + str(i) for i in range(1, df_heatmap.shape[0] + 1)]
df_heatmap.iloc[1, 2] = np.nan

plt.figure(figsize=(12, 16))

df_img_col = pd.DataFrame([f"{i:02}.jpg" for i in range(10)], columns=['Image'])
df_img_col.index = ['sample' + str(i) for i in range(1, 11)]
df_img_row = pd.DataFrame([f"{i:02}.jpg" for i in range(30)], columns=['Image'])
df_img_row.index = ["Fea" + str(i) for i in range(1, df_heatmap.shape[0] + 1)]

col_ha = HeatmapAnnotation(Image=anno_img(df_img_col,  height=16, border_width=10))
row_ha = HeatmapAnnotation(Image=anno_img(df_img_row, border_width=20), axis=0)

cm = ClusterMapPlotter(data=df_heatmap, top_annotation=col_ha, left_annotation=row_ha, col_split=2, row_split=3, col_split_gap=10,
                     row_split_gap=10,label='values',row_dendrogram=True,show_rownames=True,show_colnames=True,
                     tree_kws={'row_cmap': 'Dark2'},cmap='Spectral_r',
                       legend_gap=5,legend_hpad=2,legend_vpad=5)
plt.show()
```

example

### CODE REVIEW

**Comment by DingWB** [comment]
Hi Daisuke, Thanks for your great job. I have already merged this pull request. 
I added a very simple usage of `anno_img` to the documentation website (https://dingwb.github.io/PyComplexHeatmap/build/html/notebooks/advanced_usage.html#Image-annotation).

I would appreciate it if you could send me the jupyter notebook when you have a better example (in real biological data) for `anno_img` in the future, and I can put your example code on the documentation website.

**Comment by DingWB** [comment]
Another thing is we can add a parameter `merge` to `anno_img`, similar to what I did for `anno_label` https://github.com/DingWB/PyComplexHeatmap/blob/fd4717458f37c864fa1d340dcec8655ae2cab6bd/PyComplexHeatmap/annotations.py#L573

https://dingwb.github.io/PyComplexHeatmap/build/html/notebooks/single_cell_methylation.html
image

For example, if the user wants to plot one image for each `row_split`:
image

Basically, if `merge=True` and the path of neighbor images are the same, then we can plot only one image (such as img1 in the above example figure). 
But this would be a little complicated and I am not sure whether this is useful in practice usage.
What is your opinion?
